src/event-db/setup/graphql-setup.sql

```diff
@@ -0,0 +1,173 @@
+
+\connect CatalystEventDev;
+
+-- This is NOT part of our schema proper.  It exists to support GraphQL Admin authorization.
+
+create schema private;
+alter default privileges in schema private revoke execute ON functions FROM public;
+
+create extension if not exists "pgcrypto";
+
+-- Defined Roles
+drop role if exists "cat_admin";
+create role "cat_admin" login password 'CHANGE_ME';
+comment on role "cat_admin" IS 'Full Administrator Access to the Database.';
+GRANT "cat_admin" TO "catalyst-event-dev"; -- Make this role available to our default DB user.
+
+alter default privileges in schema public revoke EXECUTE ON FUNCTIONS FROM cat_admin;
+alter default privileges in schema private revoke EXECUTE ON FUNCTIONS FROM cat_admin;
+
+drop role if exists "cat_anon";
+create role "cat_anon" login password 'CHANGE_ME';
+comment on role "cat_anon" IS 'Unauthenticated Read Only Access to the Database.';
+GRANT "cat_anon" TO "catalyst-event-dev"; -- Make this role available to our default DB user.
+
+alter default privileges in schema public revoke EXECUTE ON FUNCTIONS FROM cat_anon;
+alter default privileges in schema private revoke EXECUTE ON FUNCTIONS FROM cat_anon;
+
+-- Allow all roles "usage" to private and public schemas
+
+grant usage on schema public  to "catalyst-event-dev", "cat_admin", "cat_anon";
+grant usage on schema private to "catalyst-event-dev", "cat_admin", "cat_anon";
+
+-- All roles can read the database
+grant select on all tables in schema public to "catalyst-event-dev", "cat_anon", "cat_admin";
+
+-- Only admin level roles can change data.
+grant insert, update, delete on all tables in schema public to "catalyst-event-dev", "cat_admin";
+
+-- Table where we store our authenticated users information.
+-- No user has direct access to this table.
+drop table if exists private.admin_account;
+create table private.admin_account (
+    id              serial primary key,
+    first_name      text not null check (char_length(first_name) < 80),
+    last_name       text check (char_length(last_name) < 80),
+    role            text,
+    about           text,
+    email           text not null unique check (email ~* '^.+@.+\..+$'),
+    password_hash   text not null,
+    created_at      timestamp default now()
+);
+
+-- Create default ADMIN user.
+
+insert into private.admin_account (first_name, last_name, role, about, email, password_hash) values
+  ('Admin', 'Default', 'cat_admin', 'Default Admin User', 'admin.default@nowhere.io', crypt('CHANGE_ME', gen_salt('bf')));
```

Comment:
should there be a larger salt?

Reply:
gen_salt generates a salt with the 'bf' type.
See: https://www.postgresql.org/docs/current/pgcrypto.html#PGCRYPTO-CRYPT-ALGORITHMS
its a 128-bit salt.